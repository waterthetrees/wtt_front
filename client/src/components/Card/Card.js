import React from 'react';
import './Card.scss';

export const Card = ({ children }) => {
  return <div className="card">{children}</div>;
};
