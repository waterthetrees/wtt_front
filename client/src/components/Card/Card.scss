.card {
  flex: 0 1 22rem;
  transition: box-shadow 0.3s ease-in-out;
  margin-bottom: 1rem;
  padding: 0 0 4px;
  height: 100%;
  background-color: whitesmoke;
  color: gray;
  break-inside: avoid;
  border-radius: 4px;

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    padding: 0 0.5rem;
    color: black;
  }

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
    color: green;
    opacity: 0.8;
  }
}
