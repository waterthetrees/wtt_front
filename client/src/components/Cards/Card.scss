.card {
  flex: 0 1 22rem;
  transition: box-shadow 0.3s ease-in-out;
  margin: 1rem;
  // padding: 2px;
  height: 100%;
  background-color: whitesmoke;
  color: gray;
  break-inside: avoid;
  border-radius: 4px;
  

  h1,
  h2,h3,h4,h5,h6 {
    padding: 0 0.5rem;
    color: black;
  }

  h2,
  .h2,.h3,.h4,.h5,.h6 {
    padding: 0 0.5rem;
    color: black;
  }

  &:focus, &:hover, &:visited, &:link, &:active {
    text-decoration: none;
    color: green;
    opacity: .8;
  }

  &__image {
    width: 100%;
    height: 100%;
    object-fit: cover;
    object-position: bottom;
  }
  
  &__image img {
    border-radius: 4px 4px 0 0;
  }

  &__item {
    color: gray;
    padding: 0 0.5rem;
  }

  &__item:first-letter { 
    font-size: 1em; 
    text-transform: uppercase;
  }

  &__item-notes {
    padding: .5rem;
    max-width: 250px;
    font-weight: 500;
  }
  
}