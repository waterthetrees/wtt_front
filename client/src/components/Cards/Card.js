import React from 'react';
import { NavLink, useParams } from 'react-router-dom';
import treeImages from '@/data/dist/treeImages.json';
import { ImageLoad } from '@/pages/Tree/TreeImage';
import './Card.scss';

export const Card = ({
  common,
  scientific,
  deciduousEvergreen,
  height,
  notes,
  // image,
  icon,
}) => {
  // const params = useParams();
  const scientificName = scientific.split(' ').join('-');
  // console.log('params', params.common, params);
  return (
    <div className="card">
      <h2>{common}</h2>
      <h4>{scientific}</h4>
      <div className="card__item">{height}</div>
      <div className="card__item">{deciduousEvergreen}</div>
      <div className="card__item-notes">{notes}</div>
      <div className="card__item-icon">{icon}</div>
      <div className="card__image">
        {scientific &&
          Object.prototype.hasOwnProperty.call(treeImages, scientific) && (
            <ImageLoad
              src={`http://localhost:3000/assets/images/treefilter/${scientificName}.jpg`}
              placeholder="placeholder.jpg"
            />
          )}
      </div>
    </div>
  );
};
