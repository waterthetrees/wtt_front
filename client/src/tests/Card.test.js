import React from 'react';
import { render } from '@testing-library/react';
import { Card } from '@/components/Card/Card';

/**
 * These snapshots are designed to catch _any_ change to UI components, so they
 * are by definition very brittle. If the changes are related to your changes
 * and look expected, update these snapshots with `npm test -- -u`
 */

describe('<Card /> spec', () => {
  it('renders Card correctly', () => {
    // const card = render(<Card />);
    // console.log('card', card);
    // // expect(card).toMatchSnapshot();
  });
});
