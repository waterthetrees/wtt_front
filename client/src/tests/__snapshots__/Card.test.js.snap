// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> spec Renders the Card component correctly 1`] = `
<div>
  <div
    class="card"
    data-testid="card"
  />
</div>
`;
